xbmc/pvr/recordings/PVRRecording.h

```diff
@@ -82,8 +84,23 @@ namespace PVR
     std::string   m_strFanartPath;    /*!< fanart path */
     unsigned      m_iRecordingId;     /*!< id that won't change while xbmc is running */
 
+  private:
     CPVRRecording(void);
     CPVRRecording(const PVR_RECORDING &recording, unsigned int iClientId);
+
+    // Prevent copy construction, even for CPVRRecording instances and friends.
+    // Note: Only declared, but intentionally not implemented
+    //       to prevent compiler generated copy ctor and to force
+    //       a linker error in case somebody tries to call it.
+    CPVRRecording(const CPVRRecording &tag);
```

Comment:
lol re "wall of text". I thought that this kind of stuff needs explanation. In another project I know very well it would be enough to just write "// n.i." behind the copy ctor and op= declaration to indicate that this intentionally.

It IS the intention that linkers will bitch! This part of the pattern. If the method shall actually and accidentally called, what we want to prevent here, the linker will be unable to find an implementation for the method and exits with an error. Problem gets detected immediately, affected code can be easily detected and fixed.